pkg/agent/manager/cache/cache.go

```diff
@@ -124,19 +125,19 @@ type StaleEntry struct {
 	ExpiresAt time.Time
 }
 
-func New(log logrus.FieldLogger, trustDomainID string, bundle *Bundle, metrics telemetry.Metrics) *Cache {
+func New(log logrus.FieldLogger, trustDomain spiffeid.TrustDomain, bundle *Bundle, metrics telemetry.Metrics) *Cache {
 	return &Cache{
-		BundleCache:  NewBundleCache(trustDomainID, bundle),
+		BundleCache:  NewBundleCache(trustDomain, bundle),
 		JWTSVIDCache: NewJWTSVIDCache(),
 
-		log:           log,
-		metrics:       metrics,
-		trustDomainID: trustDomainID,
-		records:       make(map[string]*cacheRecord),
-		selectors:     make(map[selector]*selectorIndex),
-		staleEntries:  make(map[string]bool),
+		log:          log,
+		metrics:      metrics,
+		trustDomain:  trustDomain,
+		records:      make(map[string]*cacheRecord),
+		selectors:    make(map[selector]*selectorIndex),
+		staleEntries: make(map[string]bool),
 		bundles: map[string]*bundleutil.Bundle{
```

Comment:
We should also convert this to be a `map[spiffeid.TrustDomain]*bundleutil.Bundle`

Reply:
yeah, I agree that is the best, the reason I didn't migrate it is this iteration on that file:

```
for _, federatesWith := range identity.Entry.FederatesWith {
			if federatedBundle := c.bundles[federatesWith]; federatedBundle != nil {
				w.FederatedBundles[federatesWith] = federatedBundle
```
We would need to convert `federatesWith` to a `spiffeid.TrustDomain` on every iteration. Would it be ok?

Reply:
I think it's reasonable to convert from the string in the protobuf to the go-spiffe type there. It has minimal overhead, makes the trust domains easier to work with in the code, and establishes consistency of trust domain handling across the codebase.

Reply:
👍